src/Illuminate/Database/Eloquent/Relations/HasOneOrMany.php

```diff
@@ -486,4 +486,34 @@ public function getLocalKeyName()
     {
         return $this->localKey;
     }
+
+    /**
+     * Get the foreign attributes.
+     *
+     * @return array
+     */
+    public function getForeignAttributes()
+    {
+        return [$this->getForeignKeyName() => $this->getParentKey()];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function insert(array $values)
+    {
+        if (empty($values)) {
+            return true;
+        }
+
+        if (! is_array(reset($values))) {
+            $values = [$values];
+        }
+
+        foreach ($values as $key => $value) {
+            $values[$key] = array_merge($value, $this->getForeignAttributes());
```

Comment:
should the order be swapped, giving the user the ability to override defaults if they so choose?
```php suggestion
            $values[$key] = array_merge($this->getForeignAttributes(), $value);
```

Reply:
Swapping the order is unnecessary since the user cannot manually set foreign key values in the `create` method. Also, the order should be maintained to ensure data integrity in relationships where foreign keys should be set automatically.